### PR TITLE
Currency Conversion

### PR DESCRIPTION
**Pull request by asfktz**
Related to https://github.com/eshaham/israeli-ynab-updater/issues/22

### CODE REVIEW

**Comment by eshaham** [comment]
@asfktz before I begin going over this PR, I have a few important asks and comments:
1. could you add a proper description?
2. could you try to cleanup unrelated code refactoring? it makes it very hard to follow with all the changes, and some of them seem unrelated to the changes needed for this PR.
3. I believe we can't move forward until we implement https://github.com/eshaham/israeli-bank-scrapers/issues/80. Your assumption that a transaction in a foreign currency should always be converted isn't valid. For most people, myself included, such transactions get converted automatically by the credit card company, and `chargedAmount` already holds the actual amount in `ILS` - so no conversion is needed. We will need to convert only if the new introduced `chargedCurrency` holds a foreign currency.

**Comment by asfktz** [comment]
> 1. could you add a proper description?

Done!

> 2. Could you try to clean up unrelated code refactoring? it makes it very hard to follow with all the changes, and some of them seem unrelated to the changes needed for this PR.

Got rid of the non-relevant stuff.
Please excuse me while I'm learning the art of sending small & focused PRs (:

> 3. I believe we can't move forward until we implement eshaham/israeli-bank-scrapers#80. Your assumption that a transaction in a foreign currency should always be converted isn't valid. For most people, myself included, such transactions get converted automatically by the credit card company, and chargedAmount already holds the actual amount in ILS - so no conversion is needed. We will need to convert only if the new introduced chargedCurrency holds a foreign currency.

Hmm.. yeah you're right, haven't taken that into account. sounds right 👍
